### PR TITLE
API-13186/add api config component

### PR DESCRIPTION
**Pull request by ediiotero**
Story: [API-13186](https://vajira.max.gov/browse/API-13186)

This PR add a table to the CONFIGURATION table on the Provider dashboard.

Describe the big picture of your changes here.

## Checklist

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Review by keyluck — APPROVED** [review]

File: plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      </Link>
+    </>
+  );
+};
+
+const APIConfigTable = ({ apiName, versions }: DenseTableProps) => {
+
+  const apiConfigData = versions.map(version => {
+    const majorVersion = `v${version.majorVersion}`;
+    const configObj = Object.entries(version.gatewayConfig).map( item => {
```
Comment (mheusgit) [inline plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx:38]: This appears to only display 4 of the 7 configs.  I believe this does not include Internal flag, OAS URL, or security type.

File: plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+  const handleAddClick = () => {
+    return; // TODO When access management ticket completed
+  };
+
+  const TableTitle = (): ReactElement => {
+    return (
+      <>
+        <span id="apiConfigTitle">API Configuration ({versions.length})</span>
+        <Button
```
Comment (mheusgit) [inline plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx:67]: There is no plans to support a "Add" button for the configurations.  This should be removed.

File: plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+          Add Version
+        </Button>
+      </>
+    );
+  };
+
+  return (
+    <Table
+      title={<TableTitle/>}
+      options={{ search: false, paging: false }}
```
Comment (mheusgit) [inline plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx:80]: Ticket doesn't explicitly ask for it but search should be set to "true".

File: plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx (hunk 2 above)
Comment (mheusgit) [reply plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx:67]: Also ({versions.length}) returns the number of versions instead the number of configs.

File: plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx (hunk 1 above)
Comment (ediiotero) [reply plugins/provider-dashboard/src/components/APIConfigComponent/APIConfigComponent.tsx:38]: Yeah, my fault. I was going by the image in the story I should of read the ticket again before putting in the pr. Working on these changes now.

**Comment by mheusgit** [comment]
Looks good

**Review by dsteeve — APPROVED** [review]
lgtm!

**Review by mheusgit — APPROVED** [review]
Looks good

**Pull request merged**